### PR TITLE
Issue #14937: Migrated section 7.3.1 Exception: self-explanatory methods to chapter wise testing

### PR DESCRIPTION
**Pull request by Zopsss**
#14937

### CODE REVIEW

**Comment by Zopsss** [comment]
Build is failing: 

```
[ERROR] com.puppycrawl.tools.checkstyle.internal.XdocsPagesTest.testAllStyleRules -- Time elapsed: 0.187 s <<< FAILURE!
google_style.xml rule '7.3.2 Exception: overrides' should have a test that exists
expected to be true
```

It will resolve once https://github.com/checkstyle/checkstyle/pull/15196 is merged

**Review by romani — APPROVED** [review]
Ok to merge

**Review by rdiachenko — APPROVED** [review]
lgtm

**Comment by rdiachenko** [comment]
@Zopsss please resolve conflicts

**Comment by Zopsss** [comment]
> @Zopsss please resolve conflicts

Resolved, but dont merge it. We need to merge https://github.com/checkstyle/checkstyle/pull/15196 first. @rdiachenko

File: src/xdocs/google_style.xml [hunk 1]
```diff
@@ -2290,8 +2290,8 @@
                   </a>
                 </td>
                 <td>
-                  <a href="styleguides/google-java-style-20180523/javaguide.html#s7.3.1-javadoc-exception-self-explanatory">
-                    7.3.1 Exception: self-explanatory methods</a>
+                  <a href="styleguides/google-java-style-20220203/javaguide.html#s7.3.1-javadoc-exception-self-explanatory">
+                    7.3.1 Exception: self-explanatory members</a>
```
Comment (Zopsss) [inline src/xdocs/google_style.xml:2294]: Renamed the section name, detected at: https://github.com/checkstyle/checkstyle/pull/14901#discussion_r1660928171

Comment (Zopsss) [reply src/xdocs/google_style.xml:2294]: also updated the test class name according to this

**Comment by Zopsss** [comment]
Build is failing:

```
[ERROR] Failures: 
[ERROR]   XdocsPagesTest.testAllStyleRules:1660->validateStyleModules:1789->validateModuleWiseTesting:1864 google_style.xml rule '7.3.2 Exception: overrides' module 'MissingJavadocMethod' should have a test that exists
expected to be true
```

Will be resolved once https://github.com/checkstyle/checkstyle/pull/15201 is merged

**Comment by romani** [comment]
@Zopsss , please resolve conflict

**Comment by romani** [comment]
@Zopsss , there are still bunch of failures, please address them.

**Comment by Zopsss** [comment]
@romani conflicts resolved.

**Pull request merged**